staging/src/k8s.io/apiserver/pkg/authentication/authenticatorfactory/metrics.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package authenticatorfactory
+
+import (
+	"context"
+	"sync"
+
+	compbasemetrics "k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+)
+
+type registerables []compbasemetrics.Registerable
+
+var registerMetrics sync.Once
+
+// register all metrics.
+func register() {
+	registerMetrics.Do(func() {
+		for _, metric := range metrics {
+			legacyregistry.MustRegister(metric)
+		}
+	})
+}
+
+func init() {
+	register()
+}
+
+var (
+	requestTotal = compbasemetrics.NewCounterVec(
+		&compbasemetrics.CounterOpts{
+			Name:           "apiserver_delegated_authorization_request_total",
+			Help:           "Number of HTTP requests partitioned by status code and verb.",
+			StabilityLevel: compbasemetrics.ALPHA,
+		},
+		[]string{"code", "verb"},
+	)
+
+	requestLatency = compbasemetrics.NewHistogramVec(
+		&compbasemetrics.HistogramOpts{
+			Name:           "apiserver_delegated_authorization_request_duration_seconds",
+			Help:           "Request latency in seconds. Broken down by status code and verb.",
+			Buckets:        []float64{0.25, 0.5, 0.7, 1, 1.5, 3, 5, 7},
```

Comment:
do we care whether it is 3, 5 or 7? Isn't everything above 1s already very bad?

Reply:
Do we need to break down delegated auth by verb?  What other verbs are there?

Reply:
Is latency by code even that important?

Reply:
I'm okay with dropping the code as well.

Reply:
Dropping code means not being able to tell the difference between timeouts vs slow but successful requests. They'll all just be mashed together.

Reply:
The buckets are a little strange and seem kinda arbitrary. What is the timeout for these requests? Setting a top bucket at 7s can lead to weird graphing artifacts (this was the case for `apiserver_request_latencies` (or whatever it was called back in the day), where we used to set the top bucket at 8s.

Reply:
> Dropping code means not being able to tell the difference between timeouts vs slow but successful requests. They'll all just be mashed together.

good point, we should keep the code then

Reply:
> The buckets are a little strange and seem kinda arbitrary. What is the timeout for these requests? Setting a top bucket at 7s can lead to weird graphing artifacts (this was the case for apiserver_request_latencies (or whatever it was called back in the day), where we used to set the top bucket at 8s.

What do you propose here? As Stefan already mentioned anything beyond 1s is considered slow.
Also, the default timeout (client-side) is 10s for the delegated authn/z requests.

Reply:
i'd recommend 10s then.

Reply:
done, thx.